common/src/main/java/com/genexus/db/BatchUpdateCursor.java

```diff
@@ -205,14 +205,18 @@ public boolean pendingRecords() {
 		return mPreparedStatement.getRecordCount() > 0;
 	}
 
-	public void beforeCommitEvent() throws SQLException {
-		try {
-			DynamicExecute.dynamicInstaceExecute(mPreparedStatement.getOnCommitInstance(),
+	public boolean beforeCommitEvent(java.lang.Object o) throws SQLException {
+		boolean done = false;
+		if (o == null || o == mPreparedStatement.getOnCommitInstance()) {
+			try {
+				DynamicExecute.dynamicInstaceExecute(mPreparedStatement.getOnCommitInstance(),
 					mPreparedStatement.getOnCommitMethod(), null);
-		} catch (Exception ex) {
-			throw new SQLException(ex.getMessage());
+				done = true;
+			} catch (Exception ex) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)